### ISSUE TITLE
[Sprint] Implement Optimized Feedforward Neural Network (ANN)

### ISSUE THREAD

**Issue by ankit02327** [ankit02327/stock-price#116]
As part of the Algorithmic & ML Sprint, we need to implement a production-ready, optimized Artificial Neural Network (ANN) for stock price prediction in `backend/algorithms/optimised/ann/ann.py`.

While we have a scratch implementation of a multi-layer perceptron, the optimized version should leverage `TensorFlow/Keras` to benefit from hardware acceleration, modern optimizers (Adam), and advanced regularization techniques (Dropout, Batch Normalization). Crucially, because we are predicting time-series data, we must ensure that the training process does not leak future data into the past.

**Tasks:**
- [ ] Create `ann.py` inside `backend/algorithms/optimised/ann/` and implement the `ANNModel` class inheriting from `ModelInterface`.
- [ ] Build a robust Keras `Sequential` model with `Dense`, `Dropout`, and `BatchNormalization` layers.
- [ ] Implement the `fit()` method using `shuffle=False` to preserve the chronological order of the stock data, preventing time-series data leakage.
- [ ] Implement Early Stopping callbacks to halt training when validation loss stops improving.

**Comment by ankit02327** [comment]
1. **Model Architecture:** Initialize a Keras `Sequential` model in the `fit()` method. Use a configuration like `Dense(128, relu)` -> `BatchNormalization` -> `Dropout(0.3)` -> `Dense(64, relu)` -> `Dense(1, linear)` for the regression head.
2. **Preventing Data Leakage:** When calling `model.fit()`, set `validation_split=0.2` and, most importantly, `shuffle=False`. This ensures the last 20% of the timeline is used for validation and the batch training does not mix future/past data.
3. **Scaling:** Use `sklearn.preprocessing.RobustScaler` to scale the `X` features before feeding them into the neural network, mitigating the impact of financial outliers.

**Comment by ankit02327** [comment]
```py
import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, BatchNormalization
from tensorflow.keras.callbacks import EarlyStopping
from sklearn.preprocessing import RobustScaler

# Inside ANNModel.fit(self, X, y):
self.scaler = RobustScaler()
X_scaled = self.scaler.fit_transform(X)

self.model = Sequential([
    Dense(128, activation='relu', input_shape=(X.shape[1],)),
    BatchNormalization(),
    Dropout(0.3),
    Dense(64, activation='relu'),
    BatchNormalization(),
    Dropout(0.2),
    Dense(1, activation='linear')
])

self.model.compile(optimizer='adam', loss='mse', metrics=['mae'])

early_stopping = EarlyStopping(
    monitor='val_loss', 
    patience=15, 
    restore_best_weights=True
)

self.model.fit(
    X_scaled, y,
    epochs=200,
    batch_size=32,
    validation_split=0.2,
    shuffle=False,  # Critical for time-series
    callbacks=[early_stopping],
    verbose=0
)
```

**Comment by vighnesh2005** [comment]
let me give it a try

**Comment by ankit02327** [comment]
You can put your progress here.

**Comment by ankit02327** [comment]
@vighnesh2005 I have assigned the issue.
